### PR TITLE
Add dynamic disk allocation (resolves #216)

### LINKED ISSUE #216: Add dynamic disk resource allocation for rnaseq_cgl

### PR DESCRIPTION
**Pull request by jvivian**
This PR is not ready and is being debugged.

### CODE REVIEW

**Comment by jvivian** [comment]
@hannes-ucsc — Ready for review!

File: src/toil_scripts/rnaseq_cgl/rnaseq_cgl_pipeline.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 import yaml
 from bd2k.util.files import mkdir_p
 from bd2k.util.processes import which
-from toil.job import Job
+from toil.job import Job, PromisedRequirement as PReq
```
Comment (hannes-ucsc) [inline src/toil_scripts/rnaseq_cgl/rnaseq_cgl_pipeline.py]: It looks like only line 79 is really affected by this, and it still is longer than allowed (116 < 100). For the sake of readability, I'll buy you two tacos if you use the original name …

File: src/toil_scripts/rnaseq_cgl/rnaseq_cgl_pipeline.py [hunk 2]
```diff
@@ -76,7 +76,7 @@ def preprocessing_declaration(job, config, tar_id, r1_id, r2_id):
     :param str r1_id: FileStoreID of sample read 1 (or None)
     :param str r2_id: FileStoreID of sample read 2 (or None)
     """
-    disk = '2G' if config.ci_test else '100G'
+    disk = PReq(lambda x: 3 * x.size, tar_id) if tar_id else PReq(lambda x, y: 2 * (x.size + y.size), r1_id, r2_id)
```
Comment (hannes-ucsc) [inline src/toil_scripts/rnaseq_cgl/rnaseq_cgl_pipeline.py]: And break this line up into a if statement.

**Comment by hannes-ucsc** [comment]
Done with review.

File: src/toil_scripts/rnaseq_cgl/rnaseq_cgl_pipeline.py (hunk 1 above)
Comment (jvivian) [inline src/toil_scripts/rnaseq_cgl/rnaseq_cgl_pipeline.py]: Not fair... my weakness is tacos. You got yourself a deal.

**Comment by jvivian** [comment]
@hannes-ucsc — Addressed review comments.

**Comment by hannes-ucsc** [comment]
+1

**Pull request merged**